src/py/mat3ra/made/tools/utils.py

```diff
@@ -96,3 +96,100 @@ def get_norm(vector: List[float]) -> float:
         float: The norm of the vector.
     """
     return float(np.linalg.norm(vector))
+
+
+# Condition functions:
+# TODO: Predefined equations can be exported using a factory or enum
+def _is_2d_point_in_circle(coordinate: List[float], x=0, y=0, r=1) -> bool:
+    """
+    Check if a point is inside a circle.
+    Args:
+        coordinate (List[float]): The coordinate to check.
+        x (float): The x-coordinate of the circle center.
+        y (float): The y-coordinate of the circle center.
+        r (float): The radius of the circle.
+
+    Returns:
+        Callable[[List[float]], bool]: The condition function to check if a point is inside the circle.
+    """
+    return (coordinate[0] - x) ** 2 + (coordinate[1] - y) ** 2 <= r**2
+
+
+def is_point_in_cylinder(
+    coordinate: List[float], center_position: List[float], radius: float = 0.25, min_z: float = 0, max_z: float = 1
+) -> bool:
+    """
+    Check if a point is inside a cylinder.
+    Args:
+        coordinate (List[float]): The coordinate to check.
+        center_position (List[float]): The coordinates of the center position.
+        min_z (float): Lower limit of z-coordinate.
+        max_z (float): Upper limit of z-coordinate.
+        radius (float): The radius of the cylinder.
+
+    Returns:
+        bool: True if the point is inside the cylinder, False otherwise.
+    """
+    return (
+        _is_2d_point_in_circle(coordinate, center_position[0], center_position[1], radius)
+        and min_z <= coordinate[2] <= max_z
+    )
+
+
+def _is_2d_point_in_rectangle(coordinate: List[float], x_min=0, y_min=0, x_max=1, y_max=1) -> bool:
+    """
+    Check if a point is inside a rectangle.
+    Args:
+        coordinate (List[float]): The coordinate to check.
+        x_min (float): Lower limit of x-coordinate.
+        y_min (float): Lower limit of y-coordinate.
+        x_max (float): Upper limit of x-coordinate.
+        y_max (float): Upper limit of y-coordinate.
+
+    Returns:
+        bool: True if the point is inside the rectangle, False otherwise.
+    """
+    return x_min <= coordinate[0] <= x_max and y_min <= coordinate[1] <= y_max
+
+
+def is_point_in_box(
+    coordinate: List[float], min_coordinate: List[float] = [0, 0, 0], max_coordinate: List[float] = [1, 1, 1]
+) -> bool:
+    """
+    Check if a point is inside a box.
+    Args:
+        coordinate (List[float]): The coordinate to check.
+        min_coordinate (List[float]): The minimum coordinate of the box.
+        max_coordinate (List[float]): The maximum coordinate of the box.
```

Comment:
We have to explain somewhere that all coordinates are crystal, no?

Reply:
Technically it can be cartesian or crystal, you choose that when using the function.

```
def filter_by_cylinder(
    material: Material,
    center_position: List[float] = [0.5, 0.5],
    min_z: float = 0,
    max_z: float = 1,
    radius: float = 0.25,
    use_cartesian_coordinates: bool = False, <<< can be adjusted as well as coordinates values
    invert_selection: bool = False,
) -> Material:
```